starter-code/Bookshelf/app/src/main/java/ly/generalassemb/drewmahrt/bookshelf/MainActivity.java

```diff
@@ -51,11 +55,39 @@ public long getItemId(int position) {
 
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
-                //TODO: Update the view
+                if (convertView == null) {
+                    LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                    convertView = inflater.inflate(android.R.layout.simple_list_item_2, null);
+                }
+
+                TextView  textView = (TextView) convertView.findViewById(android.R.id.text1);
+
+                textView.setText(mBookList.get(position).getTitle());
+
+                TextView textView2 = (TextView) convertView.findViewById(android.R.id.text2);
+
+                textView2.setText(mBookList.get(position).getAuthor());
+
+                return convertView;
+
+
             }
         };
 
         //TODO: Set the ListView's adapter
+        listView.setAdapter(mBookAdapter);
+        mBookAdapter.notifyDataSetChanged();
```

Comment:
No need to call `notifyDataSetChanged()` immediately after setting the adapter. You only need to call it if the data changes after the adapter has been set.